src/syncMain.ts

```diff
@@ -0,0 +1,56 @@
+import { ipcMain, webContents } from "electron";
+import {
+	Action,
+	applyMiddleware,
+	Middleware,
+	StoreCreator,
+	StoreEnhancer,
+} from "redux";
+
+import {
+	freeze,
+	preventDoubleInitialization,
+	stopForwarding,
+	validateAction,
+} from "./utils";
+
+const middleware: Middleware = (store) => {
+	ipcMain.handle("electron-redux.FETCH_STATE", async () => {
+		// Stringify the current state, and freeze it to preserve certain types
+		// that you might want to use in your state, but aren't JSON serializable
+		// by default.
+		return JSON.stringify(store.getState(), freeze);
```

Comment:
Actually I don't like idea of having this by default baked into the lib - it goes kinda against the redux recommendations itself:
> It is highly recommended that you only put plain serializable objects, arrays, and primitives into your store.
I would like to extract this and make it configurable so users when using the middleware can configure it with their own serializer at their own risk (and we ofc can provide them with example serializer used here 👍🏻 )